### ISSUE TITLE
Feature importance based on tree interpreter

### ISSUE THREAD

**Issue by sjiang87** [deephyper/deephyper#46]
1.  First I created a csv like the following for all the operations, the last column is the reward.
nas_result.csv 

> dim(4)[cell1],repeat(1)[cell1],attn(linear)[cell1],head(1)[cell1],aggr(mean)[cell1],update(mlp)[cell1],act(linear)[cell1],skip[link1],dim(16)[cell2],repeat(1)[cell2],attn(const)[cell2],head(4)[cell2],aggr(mean)[cell2],update(mlp)[cell2],act(linear)[cell2],skip[link2],skip[link3],dim(16)[cell3],repeat(1)[cell3],attn(gat)[cell3],head(1)[cell3],aggr(max)[cell3],update(mlp)[cell3],act(softplus)[cell3],connect[link4],skip[link5],skip[link6],GlobalSumPool(node),-0.3519964814186096
> dim(16)[cell1],repeat(1)[cell1],attn(gen-linear)[cell1],head(1)[cell1],aggr(sum)[cell1],update(gru)[cell1],act(elu)[cell1],connect[link1],dim(16)[cell2],repeat(1)[cell2],attn(gcn)[cell2],head(4)[cell2],aggr(mean)[cell2],update(gru)[cell2],act(tanh)[cell2],connect[link2],connect[link3],dim(8)[cell3],repeat(1)[cell3],attn(gen-linear)[cell3],head(2)[cell3],aggr(max)[cell3],update(mlp)[cell3],act(relu)[cell3],skip[link4],connect[link5],connect[link6],GlobalAvgPool(node),-0.3885207176208496

2.  Then, I used the following function

```
from treeinterpreter import treeinterpreter as ti

def feature_importance(DATA_DIR, PLOT_DIR):
    train_data = pd.read_csv(DATA_DIR + 'nas_result.csv', header=None)
    df = train_data
    df_new = pd.DataFrame()
    for i in range(df.shape[1]):
        if (df.dtypes[i] == 'object'):
            vals = pd.get_dummies(df.iloc[:,i])
        else:
            vals = df.iloc[:,i]
        df_new = pd.concat([df_new.reset_index(drop=True), vals.reset_index(drop=True)], axis=1)
    X = df_new.iloc[:, :-1]
    y = df_new.iloc[:, -1]
    scaler = StandardScaler()
    y = scaler.fit_transform(y.values[..., np.newaxis]).squeeze()
    reg = RandomForestRegressor(n_estimators=100, random_state=0).fit(X.values, y)
    prediction, bias, contributions = ti.predict(reg, X.values)
    mask = np.copy(X.values)
    mask = mask.astype(float)
    mask[mask==0] = -1
    importance = np.multiply(contributions, mask)
    importance = importance.mean(axis=0)
    importance = importance / np.max(np.abs(importance))
    indices = np.argsort(importance)[-5:] # 5 positive operations
    indices_neg = np.argsort(importance)[:5] # 5 negative operations
    plt.figure(figsize=(8,3.5))
    plt.barh(range(5, 10), importance[indices], align='center')
    plt.barh(range(5), importance[indices_neg], align='center')
    plt.yticks(range(10), [X.columns[i] for i in indices_neg] + [X.columns[i] for i in indices] )
    plt.xlabel('Relative Importance')
    plt.tight_layout()
    plt.savefig(PLOT_DIR + 'feature_importance.png', dpi=300, bbox_inches='tight')
```
